test/integration/user_cannot_view_other_users_orders_test.rb

```diff
@@ -0,0 +1,15 @@
+require 'test_helper'
+
+class UserCannotViewOtherUsersOrdersTest < ActionDispatch::IntegrationTest
+  test "user cannot view other user's orders" do
+    user1, user2 = create_list(:user, 2)
+    order1 = user1.orders.create
+    order2 = user1.orders.create
+
+    ApplicationController.any_instance.stubs(:current_user).returns(user2)
+
+    assert_raises(ActiveRecord::RecordNotFound) do |variable|
```

Comment:
Unused block argument - `variable`. You can omit the argument if you don't care about it.